src/main/java/net/sigmarik/abilitymod/mixin/PlayerEntityMixin.java

```diff
@@ -29,61 +30,107 @@
-                addStatusEffect(new StatusEffectInstance(StatusEffects.WEAKNESS, timer));
-            }
-            if (timer == 0) {
-                addStatusEffect(new StatusEffectInstance(StatusEffects.WEAKNESS, 20));
-                addStatusEffect(new StatusEffectInstance(StatusEffects.HUNGER, 20));
-                addStatusEffect(new StatusEffectInstance(StatusEffects.NAUSEA, 20));
-                addStatusEffect(new StatusEffectInstance(StatusEffects.DARKNESS, 20));
-                addStatusEffect(new StatusEffectInstance(StatusEffects.SPEED, 20));
-            }
+        if (timer < AbilityMod.ADDICTION_WARNING_TIMER) {
+            addStatusEffect(new StatusEffectInstance(StatusEffects.WEAKNESS, timer));
+        }
+        if (timer == 0) {
+            addStatusEffect(new StatusEffectInstance(StatusEffects.WEAKNESS, 20));
+            addStatusEffect(new StatusEffectInstance(StatusEffects.HUNGER, 20));
+            addStatusEffect(new StatusEffectInstance(StatusEffects.NAUSEA, 20));
+            addStatusEffect(new StatusEffectInstance(StatusEffects.DARKNESS, 20));
+            addStatusEffect(new StatusEffectInstance(StatusEffects.SPEED, 20));
         }
     }
 
     private void tickBoatMagnet() {
-        if (ServerState.hasTrait((PlayerEntity)(Object)this, AbilityMod.TRAIT_BOAT_MAGNET) && hasVehicle() &&
+        if (!ServerState.hasTrait((PlayerEntity)(Object)this, AbilityMod.TRAIT_BOAT_MAGNET) && hasVehicle() &&
                 (getVehicle().getType() == EntityType.BOAT || getVehicle().getType() == EntityType.CHEST_BOAT)) {
-            Box attractionBox = Box.of(getPos(),
+            return;
+        }
+        Box attractionBox = Box.of(getPos(),
                     AbilityMod.BOAT_ATTRACTION_DISTANCE,
                     AbilityMod.BOAT_ATTRACTION_DISTANCE,
                     AbilityMod.BOAT_ATTRACTION_DISTANCE);
 
-            final Predicate<Entity> boatPredicate = boatEntity -> true;
+        final Predicate<Entity> boatPredicate = boatEntity -> true;
 
-            for (Entity entity : getEntityWorld().getEntitiesByClass(BoatEntity.class, attractionBox, boatPredicate)) {
-                if (entity.hasPassenger(this)) continue;
-                if (entity.hasVehicle()) continue;
-                if (entity.getBoundingBox().intersects(getVehicle().getBoundingBox())) continue;
+        for (Entity entity : getEntityWorld().getEntitiesByClass(BoatEntity.class, attractionBox, boatPredicate)) {
+            if (entity.hasPassenger(this)) continue;
+            if (entity.hasVehicle()) continue;
+            if (entity.getBoundingBox().intersects(getVehicle().getBoundingBox())) continue;
 
-                entity.addVelocity(getPos().subtract(entity.getPos()).add(getVelocity())
+            entity.addVelocity(getPos().subtract(entity.getPos()).add(getVelocity())
                        .multiply(1.0, 0.0, 1.0).multiply(AbilityMod.BOAT_ATTRACTION_FACTOR));
+        }
+    }
+
+    private void tickLightBurn() {
```

Comment:
`tickLightDamage`? It corelates a bit more with the trait name.

It's just a suggestion, though.